src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -22,6 +22,7 @@ internal sealed class Host : IHost, IAsyncDisposable
         private readonly IHostEnvironment _hostEnvironment;
         private readonly PhysicalFileProvider _defaultProvider;
         private IEnumerable<IHostedService> _hostedServices;
+        private bool _stopping;
```

Comment:
volatile?

Reply:
I don't believe it is necessary, is it?

When would a read to `_stopping` be cached such that we'd need to the current value?

If a `OperationCanceledException` was thrown on a thread before the host was "Stopped", we would want that logged. The only time we don't want to log the exception is explicitly when the Host is being stopped (which sets the bool), the `OperationCanceledException` occurs (which should always read the version of `_stopping` AFAICT).

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile

> The volatile keyword indicates that a field might be modified by multiple threads that are executing at the same time.

This isn't the case here - we always expect `_stopping = true` to happen before the background task is canceled. Even if the background task is canceled/caught on a different thread.

Reply:
I usually just mark anything access across threads as volatile so the JIT forces a read from the field.

Reply:
That seems like a pretty simple rule to follow. Changing.